### PR TITLE
Adding ID Token roles mapping

### PR DESCRIPTION
**Pull request by vakninr**
* Adding fallback support to read roles from ID Token which is JWT format
* Taking the roles from the JWT based on the OIDC_ROLE_PATH_IN_RETURN
* Added test to validate the role mapping

### CODE REVIEW

File: back/user_auth/views.py [hunk 1]
```diff
@@ -322,26 +323,46 @@ def __sync_user(self, user_info):
...
             return []
+        
+    def __extract_claims_from_id_token(self):
+        id_token = self.request.session["id_token"]
+        if not id_token:
+            logger.info("OIDC: could not find ID Token to extract claims")
+            return {}
+        
+        try:
+            claims = jwt.decode(id_token,key='secret',algorithms=["HS256"])
```
Comment (GDay) [inline back/user_auth/views.py:361]: I am not super familiar with OIDC, but can we safely assume that this key is always "secret"?

Just to be on the safe side, we might want to use this to do the validation and extracting info: https://pyjwt.readthedocs.io/en/latest/usage.html#oidc-login-flow

**Review by GDay — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! Really appreciate it!

Could you fix flake8/black please? (see error from Github action)

File: back/user_auth/views.py (hunk 1 above)
Comment (vakninr) [reply back/user_auth/views.py:361]: Sure I will fix the flake8,black was not sure how to run it. Regarding the secret I will validate, but even when you decrypt jwt in jwt.io website you don't need really the secret. The secret is usually used to do validation on the JWT that it is valid and properly signed. But here I'm not concerned about security and don't really care of the signing validation just about extracting the claims out of the token.
The validation is already done using the OIDC protocols and the auth module...here we just checking the token info

**Comment by vakninr** [comment]
I updated the PR, Seems like it passed the all the isort,black and flake8. (At least locally)
I modified the function that parse the claims by changing the secret to an option that no validation will be needed. (As I wrote above it is not interesting here, we just want the claims).

**Review by GDay — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**